source/common/tracing/http_tracer_impl.cc

```diff
@@ -173,6 +174,17 @@ void HttpTracerUtility::finalizeDownstreamSpan(Span& span, const Http::HeaderMap
     span.setTag(Tracing::Tags::get().HttpProtocol,
                 AccessLog::AccessLogFormatUtils::protocolToString(stream_info.protocol()));
 
+    const auto& remote_address = stream_info.downstreamDirectRemoteAddress();
+    ASSERT(remote_address != nullptr);
```

Comment:
nit: del assert which will crash in an obvious way on the next line, same at 182 below.